knimin/static/js/platemapper.js

```diff
@@ -0,0 +1,568 @@
+  $i.appendTo($d);
+  // Store the input in the array for easy access when navigating on the
+  // plate map
+  this.wellInformation[row][column].inputTag = $i;
+  // Return the top div
+  return $d;
+};
+
+
+/**
+ *
+ * Constructs the HTML elements of the plate map
+ *
+ **/
+PlateMap.prototype._drawPlate = function() {
+  var $row, $col, $well, $table, $textArea, $btn, $span, that, study;
+  that = this;
+
+  // Add the header
+  $('<label><h3>Plate <i>' + this.name + '</i> (ID: ' + this.plateId + ') &nbsp;&nbsp;</h3></label>').appendTo(this.target);
+  // Add the buttons next to the header
+  // Save button
+  $btn = $('<button>').addClass('btn btn-info').appendTo(this.target).append(' Save');
+  $btn.click(function (e) {
+    that.savePlate();
+  });
+  $('<span>').attr('id', 'save-btn-icon').addClass('glyphicon glyphicon-save').prependTo($btn);
+  this.target.append(' ');
+  // Proceed to extraction button
+  $btn = $('<button>').addClass('btn btn-success').appendTo(this.target).append(' Extract');
+  $('<span>').addClass('glyphicon glyphicon-share').prependTo($btn);
+  $btn.click(function (e) {
+    that.extractPlate();
+  });
+  this.target.append(' ');
+  // Add the comment button. We need to add it in a span so we can have both
+  // the bootstrap tooltip and the modal triggered
+  $span = $('<span>').attr('data-toggle', 'tooltip').attr('data-placement', 'right').attr('title', 'Add well comment').attr('id', 'well-comment');
+  $span.appendTo(this.target);
+  $span.tooltip();
+  $btn =  $('<button>').addClass('btn').append(' Comment well');
+  $btn.attr('data-toggle', 'modal').attr('data-target', '#wellCommentModal').attr('id', 'comment-modal-btn').attr('pm-row', 0).attr('pm-col', 0);
+  $('<span>').addClass('glyphicon glyphicon-comment').prependTo($btn);
+  $btn.appendTo($span);
+  this.target.append(' ');
+  // Add the help button
+  $btn = $('<button>').addClass('btn btn-info').appendTo(this.target).append(' help');
+  $btn.attr('data-toggle', 'modal').attr('data-target', '#myHelpModal');
+  $('<span>').addClass('glyphicon glyphicon-info-sign').prependTo($btn);
+
+
+  // Add the plate information
+  $('</br><b>Plate type: </b>' + this.plateType + '</br>').appendTo(this.target);
+  $('<b>Created on: </b>' + this.createdOn + '</br>').appendTo(this.target);
+  $('<b>Created by: </b>' + this.createdBy + '</br>').appendTo(this.target);
+  // Add studies
+  $('<b>Studies:</b>').appendTo(this.target);
+  $.each(this.studies, function(idx, study) {
+      that.target.append(' ');
+      $('<span>').css({'background-color': PlateMap._qiimeColors[idx]}).html("&nbsp;&nbsp;&nbsp;&nbsp;").appendTo(that.target);
+      that.target.append(' ' + study.title + ' (');
+      $('<a>').attr('target', '_blank').attr('href', 'https://qiita.ucsd.edu/study/description/' + study.study_id).text('Qiita: ' + study.study_id).appendTo(that.target);
```

Comment:
would it be possible to centralize at least the base URLs here into variables? It is likely that they will be used elsewhere (or are already), and we may want to easily change them in the future.